src/Handler/TagHandler.php

```diff
@@ -117,9 +119,14 @@ public function addTags(array $tags)
      */
     public function invalidateTags(array $tags)
     {
-        $tagExpression = sprintf('(%s)(,.+)?$', implode('|', array_map('preg_quote', $this->escapeTags($tags))));
-        $headers = [$this->tagsHeader => $tagExpression];
-        $this->invalidator->invalidate($headers);
+        if (empty($tags)) {
+            return $this;
+        }
+
+        $this->invalidator->invalidateTags(
+            $this->escapeTags($tags),
```

Comment:
I'm fine with that, had the same concern.

But then we agree that we also remove `getTagsHeaderValue` to make this only focus on tags then, right?